airqo-platform/AirQo-api
# read events using Date Ranges

# read events using Date Ranges 

**_WHAT DOES THIS PR DO?_**
- [x] From data management microservice, read a Date Range of measurements in JSON format. Both recent and most recent from the optional date query parameters. Two new endpoints created for this.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[PLAT-868]

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/en-data-mgt

**_HOW DO I TEST OUT THIS PR?_**
```
cd AirQo-api/src/data-mgt/node
npm install
npm run stage-mac
```
🚨 The [.env](https://docs.google.com/document/d/1gX94v2Yavz_reD1yvC4NjJ_90gTQV7EhfckWRObeUQI/edit?usp=sharing) has also been updated accordingly. 

🚨 Also note that you need to change all the Redis cache values in your **.env** to 0 (Zero) on your local machines just to ensure that you do not have to "wait" when carrying out tests.  Otherwise, you will keep on getting the same results and think that the endpoint is not working yet you are getting cached results!

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [x] [retrieve device readings using date ranges](https://app.gitbook.com/o/-MCogOVGQqjUharJKchL/s/-MKqyQkcfs54GYi-q96G/data-management#retrieve-device-readings)
- [x] [retrieve most RECENT device readings using date ranges](https://app.gitbook.com/o/-MCogOVGQqjUharJKchL/s/-MKqyQkcfs54GYi-q96G/data-management#retrieve-the-most-recent-readings)

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_SCREENSHOTS?_**
<img width="767" alt="Screenshot 2021-12-07 at 16 52 07" src="https://user-images.githubusercontent.com/1590213/145175024-5ad783c9-cebc-4479-bdf2-67a408acaa7f.png">
<img width="870" alt="Screenshot 2021-12-07 at 16 51 14" src="https://user-images.githubusercontent.com/1590213/145175058-e259a541-0384-4683-8258-9f57b058aa57.png">





[PLAT-868]: https://airqoteam.atlassian.net/browse/PLAT-868?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
